packages/atomic/src/components/search/facets/atomic-popover/atomic-popover.tsx

```diff
@@ -0,0 +1,174 @@
+  private facetElement?: HTMLElement;
+
+  public initialize() {
+    this.searchStatus = buildSearchStatus(this.bindings.engine);
+  }
+
+  popoverOpened() {
+    if (!this.isMenuVisible) {
+      const popoverOpened = new CustomEvent('popoverOpened', {
+        detail: this.facetId,
+      });
+      document.dispatchEvent(popoverOpened);
+    }
+
+    this.isMenuVisible = !this.isMenuVisible;
+  }
+
+  @Listen('popoverOpened', {target: 'document'})
+  popOpened(event: CustomEvent) {
+    if (event.detail !== this.facetId) {
+      this.isMenuVisible = false;
+    }
+  }
+
+  renderValueButton() {
+    return (
+      <Button
+        style="square-neutral"
+        onClick={() => this.popoverOpened()}
+        part="value-button"
+        class={`value-button rounded flex box-border h-full items-center mr-1.5 p-2.5 group ${
+          this.isMenuVisible
+            ? 'selected border-primary shadow-outer-light-blue text-primary'
+            : 'hover:border-primary-light focus-visible:border-primary-light '
+        }`}
+      >
+        <span
+          title={this.facetLabel}
+          part="value-label"
+          class={`value-label truncate ${
+            this.isMenuVisible
+              ? ''
+              : 'group-hover:text-primary-light group-focus:text-primary'
+          }`}
+        >
+          {this.facetLabel}
+        </span>
+        <atomic-icon
+          part="arrow-icon"
+          class={`w-2 ml-2 ${
+            this.isMenuVisible
+              ? 'rotate-180'
+              : 'group-hover:text-primary-light group-focus:text-primary'
+          } `}
+          icon={ArrowBottomIcon}
+        ></atomic-icon>
+      </Button>
+    );
+  }
+
+  @Listen('facetInitialized')
+  injectPopoverClass(event: CustomEvent<InitPopoverEventPayload>) {
+    if (this.facetElement) {
+      return;
+    }
+
+    const facet = this.bindings.store.get(event.detail.facetType)[
+      event.detail.facetId
+    ];
+
+    if (!facet) {
+      // TODO: add error msg
+      console.error('No facet found inside the Store');
+      return;
+    }
+
+    this.facetElement = facet.element;
+    this.facetId = facet.facetId;
+    this.facetLabel = facet.label;
+    this.facetElement?.classList.add('popover-nested');
+  }
+
+  render() {
+    if (this.searchStatus.state.hasError) {
+      return <Hidden></Hidden>;
+    }
+
+    if (!this.searchStatus.state.firstSearchExecuted) {
+      return (
+        <div
+          part="placeholder"
+          aria-hidden
+          class="h-10 w-24 mr-1.5 my-2 bg-neutral animate-pulse rounded"
+        ></div>
+      );
+    }
+
+    if (this.host.children.length > 1) {
+      this.error = {
+        name: 'Too many children in popover',
+        message: 'Cannot have more than one child inside a set of popover tags',
+      };
+      return (
+        <atomic-component-error
+          element={this.host}
+          error={this.error}
+        ></atomic-component-error>
+      );
+    }
+
```

Comment:
When error is defined it should render atomic-component-error automatically.
You can move this check higher up in the process, like the connected callback or initialize method